domain/Readme.md

```diff
@@ -1,64 +1,61 @@
-1. Standard JPA querying
- * One can obtain an EntityManager by using the @PersistentContext annotation on a field in a Spring bean.
- * The EntityManager can be used to write queries.
- * All Queries should be contained in the same module as the domain object that it is querying, unless the module is a plugin module.
- * **Examples: (See tests for example usage)**
-  * [MetadataStatisticsQueries](src/main/java/org/fao/geonet/repository/statistic/MetadataStatisticsQueries.java)
+1. Query methods
+   - You can create methods in the ...Repository interface and the method name will be parsed by Spring Data and a query will be created for you based on the name.  There are many different options including in, and, or ways to write the methods.
+   - While the methods are easy to write, they are not very flexible.  Using Specifications tends to be a more flexible solution and is therefore preferred.
+   - **Examples: (See tests for example usage)**
+     * [UserRepository](src/main/java/org/fao/geonet/repository/UserRepository.java)
+     * [OperationAllowedRepository](src/main/java/org/fao/geonet/repository/OperationAllowedRepository.java)
+2. Specifications
+   - The Specification interface provides a composable API for writing queries.
+   - For example, you could write a hasMetadataId specification and a hasOwnerId specification.  One could then compose them using not, and, or.
+   - Each specification should be in the appropriate <DomainObject>Specs class.
+   - Each specification is to be created with a static method.
+   - Each specification must have a test.
+   - **Examples: (See tests for example usage)**
+     * [MetadataSpecs](src/main/java/org/fao/geonet/repository/specification/MetadataSpecs.java)
+     * [UserSpecs](src/main/java/org/fao/geonet/repository/specification/UserSpecs.java)
+3. Custom Queries
+   - Spring JPA Repositories can have custom queries with very custom implementations.  A Repository will extend an interface with the custom methods and the implementations would be in a subclass of the interface.
+   - The implementation will use the normal EntityManager (JPA) API for constructing the queries.
+   - **Examples: (See tests for example usage)**
+     * [MetadataRepositoryCustom](src/main/java/org/fao/geonet/repository/MetadataRepositoryCustom.java)
+     * [MetadataRepositoryImpl](src/main/java/org/fao/geonet/repository/MetadataRepositoryImpl.java)
+4. Standard JPA querying
+   - One can obtain an EntityManager by using the @PersistentContext annotation on a field in a Spring bean.
+   - The EntityManager can be used to write queries.
+   - All Queries should be contained in the same module as the domain object that it is querying, unless the module is a plugin module.
+   - **Examples: (See tests for example usage)**
+     * [MetadataStatisticsQueries](src/main/java/org/fao/geonet/repository/statistic/MetadataStatisticsQueries.java)
```

Comment:
This class doesn't exist anymore, I guess this one can be used for the example: https://github.com/geonetwork/core-geonetwork/blob/main/domain/src/main/java/org/fao/geonet/repository/reports/MetadataReportsQueries.java

Reply:
Changed MetadataStatisticsQueries.java to MetadataReportsQueries.java